WalletWasabi.Tests/UnitTests/Wallet/WalletSynchronizationTests.cs

```diff
@@ -43,7 +43,7 @@ public async Task InternalAddressReuseNoBlockOverlapTestAsync()
 		await realWallet.WalletFilterProcessor.StartAsync(testDeadlineCts.Token);
 
 		await realWallet.PerformSynchronizationAsync(SyncType.Turbo, testDeadlineCts.Token);
-		Assert.Single(realWallet.GetAllCoins());
+		Assert.Equal(2, realWallet.GetAllCoins().Count());
```

Comment:
What happens in this test is really interesting: The test fails with my PR without this change because, during pre-fetch of block 104 containing the 3rd transaction, the block 102 containing the first one was not processed yet, so it can't be known that the internal key was already used. Therefore it is still in the `Turbo` set of keys when processing the next block, which makes the wallet filter processor to encounter the transaction earlier than currently on master, where it will only discover it while testing the `NonTurbo` set. Actually, it's not a problem.